xbmc/platform/linux/input/LibInputKeyboard.cpp

```diff
@@ -222,55 +333,94 @@ void CLibInputKeyboard::ProcessKey(libinput_event_keyboard *e)
   if (!m_ctx || !m_keymap || !m_state)
     return;
 
-  XBMC_Event event = {};
-
   const uint32_t xkbkey = libinput_event_keyboard_get_key(e) + 8;
+  const xkb_keysym_t keysym = xkb_state_key_get_one_sym(m_state.get(), xkbkey);
   const bool pressed = libinput_event_keyboard_get_key_state(e) == LIBINPUT_KEY_STATE_PRESSED;
+  xkb_state_update_key(m_state.get(), xkbkey, pressed ? XKB_KEY_DOWN : XKB_KEY_UP);
 
-  event.type = pressed ? XBMC_KEYDOWN : XBMC_KEYUP;
-  xkb_state_update_key(m_state, xkbkey, pressed ? XKB_KEY_DOWN : XKB_KEY_UP);
-
-  const xkb_keysym_t keysym = xkb_state_key_get_one_sym(m_state, xkbkey);
+  bool flushComposer{false};
+  if (pressed && SupportsKeyComposition())
+  {
+    xkb_compose_state_feed(m_composedState.get(), keysym);
+    const xkb_compose_status composeStatus = xkb_compose_state_get_status(m_composedState.get());
+    if (composeStatus == XKB_COMPOSE_COMPOSING)
```

Comment:
switch/case maybe? Are there any other enums not handled by this if/else block?

Reply:
There's XKB_COMPOSE_NOTHING. Maybe the switch case will make this more readable indeed.